src/app/components/login/login.component.html

```diff
@@ -2,27 +2,74 @@
   <div class="row justify-content-center mt-5">
     <h2 class="text-center mb-4">Bienvenido:</h2>
     <form>
-      <div class="form-group">
-        <label>Usuario:</label>
+      <mat-form-field class="example-full-width">
+        <mat-label>Usuario</mat-label>
         <input
           type="text"
-          class="form-control"
-          id="usuario"
-          placeholder="Ingresa tu usuario"
+          matInput
+          [formControl]="userFormControl"
+          [errorStateMatcher]="matcher"
+          placeholder="Ex. JoseManuel23"
         />
-      </div>
-      <div class="form-group mt-3">
-        <label>Contraseña:</label>
+        <mat-hint>Usuario valido</mat-hint>
+        <mat-error
+          *ngIf="
+            userFormControl.hasError('minlength') &&
+            userFormControl.hasError('pattern') &&
+            !userFormControl.hasError('required')
+          "
+        >
```

Comment:
```suggestion
        <mat-error
          *ngIf="showUserError()"
        >
```
Extract this logic to a function in the controller